CUDADataFormats/TrackingRecHit/interface/TrackingRecHit2DHeterogeneous.h

```diff
@@ -226,16 +294,18 @@ TrackingRecHit2DHeterogeneous<Traits>::TrackingRecHit2DHeterogeneous(
 
   m_store16 = Traits::template make_unique<uint16_t[]>(nHits * n16, stream);
   m_store32 = Traits::template make_unique<float[]>(nHits * n32, stream);
-  m_PhiBinnerStore = Traits::template make_unique<TrackingRecHit2DSOAView::PhiBinner>(stream);
-  m_AverageGeometryStore = Traits::template make_unique<TrackingRecHit2DSOAView::AverageGeometry>(stream);
+  m_PhiBinnerStore = Traits::template make_unique<typename TrackingRecHit2DSOAView::PhiBinner>(stream);
+  m_AverageGeometryStore = Traits::template make_unique<typename TrackingRecHit2DSOAView::AverageGeometry>(stream);
 
   view->m_averageGeometry = m_AverageGeometryStore.get();
   view->m_hitsModuleStart = m_hitsModuleStart;
 
   //store transfer
   if constexpr (std::is_same_v<Traits, cms::cudacompat::GPUTraits>) {
-    cms::cuda::copyAsync(m_store16, store16, stream);
-    cms::cuda::copyAsync(m_store32, store32, stream);
+    cudaCheck(cudaMemcpyAsync(
+        m_store16.get(), store16, sizeof(uint16_t) * static_cast<int>(n16 * nHits), cudaMemcpyDefault, stream));
+    cudaCheck(cudaMemcpyAsync(
+        m_store32.get(), store32, sizeof(float) * static_cast<int>(n32 * nHits), cudaMemcpyDefault, stream));
```

Comment:
why do you go back to `cudaMemcpyAsync` ?

Reply:
Ah, I see.

Where does the input come from ?
Would it be possible to change it to be a `cms::cuda::host::unique_ptr<uint16_t[]>` or `cms::cuda::host::unique_ptr<float[]>` ?

Reply:
Not implemented? Is this still okay?

Reply:
At the end I did not implant it here because it was going slightly out of the scope of the PR. Anyway all of these constructors will become obsolete with a soon coming update to use the new `SoALayout` paradigm.